File: game/salvage.go
package game

type Salvage struct {
	MapObject
	Cargo Cargo `json:"cargo,omitempty"`
}

// create a new salvage object
func newSalvage(playerNum int, position Vector, cargo Cargo) Salvage {
	return Salvage{
		MapObject: MapObject{
			Type:      MapObjectTypeSalvage,
			Position:  position,
			PlayerNum: playerNum,
		},
		Cargo: cargo,
	}
}

File: frontend/src/lib/types/Salvage.ts
import { MapObjectTypeSalvage, type SalvageIntel } from './cs';

export function newSalvage(): SalvageIntel {
	return {
		type: MapObjectTypeSalvage,
		name: '',
		position: { x: 0, y: 0 },
		cargo: {},
		num: 0,
		playerNum: 0,
		tags: {},
		reportAge: 0
	};
}
